src/components/DateForm.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  type TextInputRefType,
+  Fieldset,
+  TextInput,
+  Notification,
+} from "@nypl/design-system-react-components"
+import type { SyntheticEvent, MutableRefObject, Dispatch } from "react"
+import { debounce } from "underscore"
+
+type DateFormPropsType = {
+  inputRef?: MutableRefObject<TextInputRefType>
+  dateAfter: string
+  dateBefore: string
+  debounceInterval: number
+  setDateRangeError: Dispatch<React.SetStateAction<boolean>>
+  changeHandler: (e: SyntheticEvent) => void
+  displayDateRangeErrorNotification: boolean
+}
+
+const DateForm = ({
```

Comment:
Not reviewing this yet because of my comment in `pages/search/advanced.tsx`. The component code here and in `SearchFilters/DateForm.tsx` have a few differences if you need to define the final place where it should go.